### PR TITLE
TOF-FBP reconstruction

### PR DESCRIPTION
**Pull request by kamilrakoczy**
Continuation of #130 

Adds TOF-FBP, attenuation correction, remapping to single layer, sparse matrix implementation, ability to save calculated sinogram.

### CODE REVIEW

**Comment by grey277** [comment]
Tests are not passing because of not existing j-pet-mlem, I will update this PR after accepting PR #144

**Comment by grey277** [comment]
retest this please

**Comment by kamilrakoczy** [comment]
retest this please

File: ImageReconstruction/tests/SinogramCreatorToolsTest.cpp [hunk 1]
```diff
@@ -85,4 +95,138 @@ BOOST_AUTO_TEST_CASE(test_lor_slice) {
       0.f, EPSILON);
 }
 
-BOOST_AUTO_TEST_SUITE_END()
+/*
```
Comment (wkrzemien) [inline ImageReconstruction/tests/SinogramCreatorToolsTest.cpp]: Why is it commented out? It is not working or is obsolete? Then maybe remove it.

File: ImageReconstruction/SinogramCreatorToolsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                      kEPSILON);
+  BOOST_REQUIRE_CLOSE(SinogramCreatorTools::getPolyFit({std::sqrt((9.57 * 9.57) + (-0.87 * -0.87)), -std::abs(2.49)}), 7.197118953414579e-01,
+                      kEPSILON);
+  BOOST_REQUIRE_CLOSE(SinogramCreatorTools::getPolyFit({std::sqrt((-11.83 * -11.83) + (2.66 * 2.66)), -std::abs(-2.72)}), 5.820874582833866e-01,
+                      kEPSILON);
+  BOOST_REQUIRE_CLOSE(SinogramCreatorTools::getPolyFit({std::sqrt((1.84 * 1.84) + (-8.89 * -8.89)), -std::abs(-4.77)}), 7.430144030486940e-01,
+                      kEPSILON);
+}
+
+BOOST_AUTO_TEST_SUITE_END()
```
Comment (wkrzemien) [inline ImageReconstruction/SinogramCreatorToolsTest.cpp]: add endline pls

**Comment by wkrzemien** [comment]
Could you add a small description what getSplitRangeNumber() does in CreatorTools.

File: ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHann.h [hunk 3]
```diff
@@ -10,24 +10,27 @@
...
+    return pos < fCutOff ? 0.5 + 0.5 * std::cos((2. * M_PI * pos) / fCutOff) : 0.;
+  }
 
 private:
-  double threshold = 0.9;
+  double fCutOff = 1.0;
 };
 
-#endif /*  !_JPetFilterRamLak_H_ */
+#endif /*  !_JPetFilterHann_H_ */
```
Comment (wkrzemien) [inline ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHann.h]: endline missing

File: ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHamming.h [hunk 4]
```diff
@@ -24,11 +24,14 @@
 class JPetFilterHamming : public JPetFilterInterface {
 public:
   JPetFilterHamming() {}
-  explicit JPetFilterHamming(double alphaValue) { alpha = alphaValue; }
-  virtual double operator()(double radius) override { return radius * (alpha + (1. - alpha * std::cos(2. * M_PI * radius))); }
+  explicit JPetFilterHamming(double maxCutOff) : fCutOff(maxCutOff) {}
+  virtual double operator()(double pos) override
+  {
+    return pos < fCutOff ? 0.54 + 0.46 * std::cos((2. * M_PI * pos) / fCutOff) : 0.;
```
Comment (wkrzemien) [inline ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHamming.h]: What are those numbers 0.54 and 0.46? Coud you add a comment, link or something and maybe define const

Comment (kamilrakoczy) [reply ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHamming.h]: They are alpha parameter for Hamming filter in windowing function, I extracted them to const value.

File: ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHann.h (hunk 3 above)
Comment (kamilrakoczy) [reply ImageReconstruction/Reconstruction/JPetRecoImageTools/JPetFilterHann.h]: done

File: ImageReconstruction/SinogramCreatorToolsTest.cpp (hunk 2 above)
Comment (kamilrakoczy) [reply ImageReconstruction/SinogramCreatorToolsTest.cpp:193]: done

File: ImageReconstruction/tests/SinogramCreatorToolsTest.cpp (hunk 1 above)
Comment (kamilrakoczy) [reply ImageReconstruction/tests/SinogramCreatorToolsTest.cpp]: I think they were commented out just to speed up local testing, I re-enabled them.

**Pull request merged**